### ISSUE TITLE
Fields contained in row field inside localized group don't get saved

### ISSUE THREAD

**Issue by ivankyrpa** [payloadcms/payload#8405]
### Link to reproduction

https://github.com/ivankyrpa/payload-cms-beta108

### Environment Info

```text
Binaries:
  Node: 20.17.0
  npm: 10.8.2
  Yarn: 1.22.22
  pnpm: 9.10.0
Relevant Packages:
  payload: 3.0.0-beta.108
  next: 15.0.0-canary.160
  @payloadcms/db-postgres: 3.0.0-beta.108
  @payloadcms/email-nodemailer: 3.0.0-beta.108
  @payloadcms/graphql: 3.0.0-beta.108
  @payloadcms/next/utilities: 3.0.0-beta.108
  @payloadcms/plugin-cloud: 3.0.0-beta.108
  @payloadcms/richtext-lexical: 3.0.0-beta.108
  @payloadcms/translations: 3.0.0-beta.108
  @payloadcms/ui/shared: 3.0.0-beta.108
  react: 19.0.0-rc-5dcb0097-20240918
  react-dom: 19.0.0-rc-5dcb0097-20240918
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.4.0: Wed Feb 21 21:44:54 PST 2024; root:xnu-10063.101.15~2/RELEASE_ARM64_T6030
  Available memory (MB): 18432
  Available CPU cores: 11
```

### About
Fields inside `row` field that is in a localized `group` field don't get saved. Probably this is tied with the issue https://github.com/payloadcms/payload/issues/8322.

### Demo

https://github.com/user-attachments/assets/d846e368-7793-4392-84c1-04558f9c7b70

### Reproduction Steps

I used `npx create-payload-app@beta` command to create repro repo. Please use `link to reproduction` for the full example.

Test collection:
```JS
export const Tests: CollectionConfig = {
  slug: 'tests',
  fields: [
    {
      name: 'localizedGroup',
      type: 'group',
      localized: true,
      fields: [
        {
          type: 'row',
          fields: [
            {
              name: 'text',
              type: 'text',
            },
            ...
          ]
        }
      ]
    },
```

Payload config:
```JS
export default buildConfig({
  collections: [Tests, ...],
  localization: {
    defaultLocale: 'en',
    locales: ['en', 'de'],
    fallback: true,
  },
  ...
})
```

### Adapters and Plugins

db-postgres

**Comment by HappyEmu** [comment]
I just upgraded my app to beta.110 after one of my reported bugs with respect to localization (and versions) in tandem with postgres was fixed: https://github.com/payloadcms/payload/issues/7695 
Since I wanted to do the switch from MongoDB to Postgres, I checked the issues for more problems related to locales and versions. This issue popped up.

TLDR: I tested this on beta.110 and this bug is still present in that version.
